src/prefs/DirectoriesPrefs.cpp

```diff
@@ -447,17 +450,43 @@ bool DirectoriesPrefs::Validate()
          wxOK | wxCENTRE | wxICON_INFORMATION);
    }
 
+   const wxString openPathString = mOpenText->GetValue();
+   const wxString savePathString = mSaveText->GetValue();
+   const wxString importPathString = mImportText->GetValue();
+   const wxString exportPathString = mExportText->GetValue();
    const wxString macroPathString = mMacrosText->GetValue();
 
-   if (!macroPathString.empty())
+   //flag for checking if atleast one directory write protected 
+   //will not be 0 if any of the paths are not writable
+   int flag = 0;
+
+   if (!macroPathString.empty() || !openPathString.empty() || !savePathString.empty() ||
+       !importPathString.empty() || !exportPathString.empty())
    {
+      const wxFileName openPath { openPathString };
+      const wxFileName savePath { savePathString };
+      const wxFileName importPath { importPathString };
+      const wxFileName exportPath { exportPathString };
       const wxFileName macroPath { macroPathString };
 
-      if (macroPath.DirExists())
-      {
-         if (!FileNames::WritableLocationCheck(macroPathString))
-            return false;
-      }
+      //Checks if the location for Open,Save.Import,Export and Macro Output has write permissions(When OK is clicked)
+      if (openPath.DirExists() && !FileNames::WritableLocationCheck(openPathString, 
+                                          wxT("Preference for Open cannot be set,")))
```

Comment:
Here are those disctinct strings I asked for above.  Again "Preference for" is unnecessary words.